agents/aliyun/fence_aliyun.py

```diff
@@ -153,18 +165,27 @@ def main():
 
 	run_delay(options)
 
-	if "--region" in options:
-		region = options["--region"]
-		if "--access-key" in options and "--secret-key" in options:
-			access_key = options["--access-key"]
-			secret_key = options["--secret-key"]
-			conn = client.AcsClient(access_key, secret_key, region)
-		elif "--ram-role" in options:
-			ram_role = options["--ram-role"]
-			role = EcsRamRoleCredential(ram_role)
-			conn = client.AcsClient(region_id=region, credential=role)
+	if "--region" not in options or options["--region"] == "":
```

Comment:
You need to change the required field for the `region` parameter, as this test would break some of the actions that doesnt require any of the required parameters (e.g. metadata).

Reply:
Yes, I was ill-conceived, I restored this code